docs/notebooks/feedback_prod.ipynb

```diff
@@ -0,0 +1,333 @@
+    "        st.session_state.messages = st.session_state.messages = []\n",
+    "\n",
+    "\n",
+    "def main():\n",
+    "    st.session_state[\"session_id\"] = \"123abc\"\n",
+    "    init_chat_history()\n",
+    "    display_chat_messages()\n",
+    "    get_and_process_prompt()\n",
+    "\n",
+    "\n",
+    "if __name__ == \"__main__\":\n",
+    "    main()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "vscode": {
+     "languageId": "plaintext"
+    }
+   },
+   "source": [
+    "You can run this with `streamlit run chatbot.py`.\n",
+    "\n",
+    "Now, you can interact with this application and click the feedback buttons after each response. \n",
+    "Visit the Weave UI to see the attached feedback.\n",
+    "\n",
+    "## Explanation\n",
+    "\n",
+    "If we consider our decorated prediction function as:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import weave\n",
+    "\n",
+    "weave.init(\"feedback-example\")\n",
+    "\n",
+    "\n",
+    "# highlight-next-line\n",
+    "@weave.op\n",
+    "def predict(input_data):\n",
+    "    # Your prediction logic here\n",
+    "    some_result = \"hello world\"\n",
+    "    return some_result"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can use it as usal to deliver some model response to the user:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "with weave.attributes(\n",
+    "    {\"session\": \"123abc\", \"env\": \"prod\"}\n",
+    "):  # attach arbitrary attributes to the call alongside inputs & outputs\n",
+    "    result = predict(input_data=\"your data here\")  # user question through the App UI"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "To attach feedback, you need the `call` object, which is obtained by using the `.call()` method *instead of calling the function as normal*:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "result, call = predict.call(input_data=\"your data here\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This call object is needed for attaching feedback to the specific response.\n",
+    "After making the call, the output of the operation is available using `result` above."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "call.feedback.add_reaction(\"👍\")  # user reaction through the App UI"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Retrieving Feedback "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can pull feedback and their associated calls using weaves export APIs. "
```

Comment:
weave => Weave's. Also not sure I would call these `export` APIs vs. something like "query APIs", or maybe just saying "Weave's Python client".